packages/driver/src/dom/elements.ts

```diff
@@ -889,77 +909,49 @@ const getActiveElByDocument = (doc: Document): HTMLElement | null => {
 }
 
 const getFirstParentWithTagName = ($el, tagName) => {
-  // return null if undefined or we're at body/html/document
-  // cuz that means nothing has fixed position
-  if (isUndefinedOrHTMLBodyDoc($el) || !tagName) {
-    return null
-  }
-
-  // if we are the matching element return ourselves
-  if (getTagName($el[0]) === tagName) {
-    return $el
-  }
+  return findParent($el.get(0), (node) => {
+    if (getTagName(node) === tagName) {
+      return node
+    }
 
-  // else recursively continue to walk up the parent node chain
-  return getFirstParentWithTagName($el.parent(), tagName)
+    return null
+  })
 }
 
 const getFirstFixedOrStickyPositionParent = ($el) => {
-  // return null if we're undefined or at not a normal DOM el
-  // cuz that means nothing has fixed position
-  if (isUndefinedOrHTMLBodyDoc($el)) {
```

Comment:
Because we think that this line doesn't make sense, @43081j is going to review

Reply:
you were right about this. `findParent` does return null when it hits a document, but doesn't handle when the initial element might be a document (or undefined).

so we still need these checks.